sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -2062,6 +2067,16 @@ def write_pipeline_spec_to_file(
             f'The output path {package_path} should end with ".yaml".')
 
 
+def _merge_pipeline_config(pipelineConfig: pipeline_config.PipelineConfig,
+                           platformSpec: pipeline_spec_pb2.PlatformSpec):
+    pipeline_config_json = json_format.ParseDict(
+        {'pipelineConfig': {
+            'pipelineTtl': pipelineConfig.get_ttl(),
+        }}, platformSpec.platforms['kubernetes'])
```

Comment:
@chensun the SDK code part of this PR correctly compiles and generates the Yaml IR as per the requirements we decided upon [here.](https://github.com/kubeflow/pipelines/pull/11112#issuecomment-2403597690)

Code: https://gist.github.com/rimolive/e0c31ccad1f6c963ec010c455b424afe
resulting IR yaml: https://gist.github.com/rimolive/fc900c8172e661a470a6890555159f97

Could you confirm that the hard coded specification of `kubernetes` platform here would work? Does the platform key in the IR always stay the same, i.e., `kubernetes`, or is it customized (to have it set as `vertex`, for example)? If yes, then we can look into having this key dynamically specified and inserted here.

Reply:
Not sure I fully understand the question. The key is hard coded for individual platforms, and the platform backend is expected to look after the key they recognize and ignore the ones they don't. For instance, the KFP backend extracts only `kubernetes` key: 

https://github.com/kubeflow/pipelines/blob/6a35ee5144fba1c72badc7c52161d8a49f34804e/backend/src/apiserver/template/v2_template.go#L72-L78

The specific feature you're adding here, TTL, doesn't apply to Vertex. So what you have in the code and IR links look good to me.

Reply:
Thank you for clarifying @chensun !